packages/sink-mongo/src/mongo.ts

```diff
@@ -0,0 +1,107 @@
+import { Sink, type SinkCursorParams } from "@apibara/indexer";
+import type { Cursor } from "@apibara/protocol";
+import type { ClientSession, DbOptions, MongoClient } from "mongodb";
+import { MongoSinkTransactionDb } from "./transaction";
+
+export interface MongoSinkOptions {
+  client: MongoClient;
+  dbName: string;
+  dbOptions?: DbOptions;
+  collections: string[];
+}
+
+export class MongoSink extends Sink {
+  constructor(
+    private client: MongoClient,
+    private config: Omit<MongoSinkOptions, "client">,
+  ) {
+    super();
+  }
+
+  async transaction(
+    { cursor, endCursor, finality }: SinkCursorParams,
+    cb: (params: {
+      db: MongoSinkTransactionDb;
+      session: ClientSession;
+    }) => Promise<void>,
+  ): Promise<void> {
+    await this.client.withSession(async (session) =>
+      session.withTransaction(async (session) => {
+        const db = this.client.db(this.config.dbName, this.config.dbOptions);
+        await cb({
+          db: new MongoSinkTransactionDb(db, session, endCursor),
+          session,
+        });
+        return "Transaction committed.";
```

Comment:
Is this needed?

Reply:
i was just testing if adding this resolves some errors XD.
not required.